Zend/zend_vm_def.h

```diff
@@ -9709,12 +9709,30 @@ ZEND_VM_HANDLER(167, ZEND_COPY_TMP, TMPVAR, UNUSED)
 	ZEND_VM_NEXT_OPCODE();
 }
 
-ZEND_VM_HANDLER(202, ZEND_CALLABLE_CONVERT, UNUSED, UNUSED)
+ZEND_VM_HANDLER(202, ZEND_CALLABLE_CONVERT, UNUSED, UNUSED, NUM|CACHE_SLOT)
 {
 	USE_OPLINE
 	zend_execute_data *call = EX(call);
 
-	zend_closure_from_frame(EX_VAR(opline->result.var), call);
+	if (opline->extended_value != (uint32_t)-1) {
+		zend_object *closure = CACHED_PTR(opline->extended_value);
+		if (closure) {
+			ZVAL_OBJ_COPY(EX_VAR(opline->result.var), closure);
+		} else {
+			closure = zend_hash_find_ptr(&EG(callable_convert_cache), call->func->common.function_name);
```

Comment:
We might probably use the function ptr here instead of the function name, which would result in a slightly faster comparison. But not crucial, as this is the slow path anyway.

Reply:
This cache doesn't apply to closures, only to FCCs to named functions.

Reply:
Oh, of course. Got mixed up with the discussions in #19754 :)